### DISCUSSION TITLE
How to do collision check?

### DISCUSSION THREAD

**Discussion by JKevin9** [isaac-sim/IsaacLab#1598] [Q&A]
Hi,

I want to check if my robot overlaps with the terrain mesh. After each time the robot dies, it respawns at a random position. However, due to my complex terrain (which includes tables, walls, and pillars), the robot might spawn inside the terrain geometry. I want to perform a collision check at each reset - if the randomly sampled position would cause the robot to be embedded in the terrain, I want to resample the position. Is there a method to achieve this?

Thanks in advance for your responses

**Comment by RandomOakForest** [comment]
Thank you for posting this. A possible solution may be to [use a ray caster sensor](https://isaac-sim.github.io/IsaacLab/main/source/overview/sensors/ray_caster.html#ray-caster). The team is currently working on better ways to do this.
